docs/DeveloperGuide.adoc

```diff
@@ -767,20 +896,91 @@ Use case ends.
 +
 Use case resumes at step 2.
 
-{More to be added}
+=== Use case:  History
+
+*MSS*
+
+1.  User requests to list the history of commands
+2.  AddressBook lists the history of commands
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There are no previous commands
++
+Use case ends.
+
+=== Use case:  Undo
+
+*MSS*
+
+1.  User requests to undo the previous command
+2.  AddressBook undoes the previous command
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The previous command is not an undoable command (add, delete, edit, clear).
++
+Use case ends.
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Should be able backwards compatible with data stored by earlier version.
+.  Should come up with automated Junit tests.
```

Comment:
this sounds functional or at least ambiguous. maybe change to the software should be well tested?